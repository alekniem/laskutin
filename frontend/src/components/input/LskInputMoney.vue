<template>
    <ValidationProvider
        :customMessages="{ regex: trans('Invalid money amount') }"
        :rules="{ regex: /^-*\d+(,\d{0,2}){0,1}$/ }"
        :vid="vid"
        ref="provider"
        tag="div"
        #default="{ errors, valid }"
    >
        <v-text-field
            @change="$emit('input', $event)"
            :error-messages="errors"
            :id="cid"
            :label="trans(label)"
            :success="valid"
            :value="value"
        />
    </ValidationProvider>
</template>

<script>
import { ValidationProvider } from "vee-validate";

export default {
    name: "LskInputMoney",

    components: {
        ValidationProvider,
    },

    props: {
        cid: { type: String, required: true },
        label: { type: String, required: true },
        value: { type: [String, Number], required: true },
        vid: { type: String, required: true },
    },
};
</script>
