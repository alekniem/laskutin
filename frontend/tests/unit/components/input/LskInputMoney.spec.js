import flushPromises from "flush-promises";
import LskInputMoney from "@/components/input/LskInputMoney";
import { createWrapper } from "UT/lsk_test_helper";

describe("LskInputMoney", () => {
    it("should match snapshot", async () => {
        const wrapper = createWrapper(LskInputMoney, {
            propsData: {
                cid: "TEST_CID",
                label: "TEST LABEL",
                value: "123,45",
                vid: "TEST_VID",
            },
        });
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot("valid input");
    });

    it("should show an error message", async () => {
        const wrapper = createWrapper(LskInputMoney, {
            propsData: {
                cid: "TEST_CID",
                label: "TEST LABEL",
                value: "INVALID",
                vid: "TEST_VID",
            },
        });
        await flushPromises();
        wrapper.vm.$refs.provider.validate();
        await flushPromises();
        expect(wrapper.find(".v-messages__message").text()).toBe("Invalid money amount");
    });
});
