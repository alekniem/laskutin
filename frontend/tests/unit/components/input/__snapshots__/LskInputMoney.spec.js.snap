// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LskInputMoney should match snapshot: valid input 1`] = `
<div>
  <div class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field success--text">
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-text-field__slot">
          <label for="TEST_CID"
                 class="v-label v-label--active theme--light success--text"
                 style="left: 0px; position: absolute;"
          >
            TEST LABEL
          </label>
          <input id="TEST_CID"
                 type="text"
          >
        </div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages theme--light success--text">
          <transition-group-stub tag="div"
                                 name="message-transition"
                                 class="v-messages__wrapper"
          >
          </transition-group-stub>
        </div>
      </div>
    </div>
  </div>
</div>
`;
